src/pkg/cli/client/byoc/gcp/stream.go

```diff
@@ -231,28 +237,40 @@ func (s *SubscribeStream) AddJobExecutionUpdate(executionName string) {
 	s.query.AddJobExecutionUpdateQuery(executionName)
 }
 
-func (s *SubscribeStream) AddJobStatusUpdate(project, etag string, services []string) {
-	s.query.AddJobStatusUpdateRequestQuery(project, etag, services)
-	s.query.AddJobStatusUpdateResponseQuery(project, etag, services)
+func (s *SubscribeStream) AddJobStatusUpdate(stack, project, etag string, services []string) {
+	s.query.AddJobStatusUpdateRequestQuery(stack, project, etag, services)
+	s.query.AddJobStatusUpdateResponseQuery(stack, project, etag, services)
 }
 
-func (s *SubscribeStream) AddServiceStatusUpdate(project, etag string, services []string) {
-	s.query.AddServiceStatusRequestUpdate(project, etag, services)
-	s.query.AddServiceStatusReponseUpdate(project, etag, services)
-	s.query.AddComputeEngineInstanceGroupInsertOrPatch(project, etag, services)
+func (s *SubscribeStream) AddServiceStatusUpdate(stack, project, etag string, services []string) {
+	s.query.AddServiceStatusRequestUpdate(stack, project, etag, services)
+	s.query.AddServiceStatusReponseUpdate(stack, project, etag, services)
+	s.query.AddComputeEngineInstanceGroupInsertOrPatch(stack, project, etag, services)
 	s.query.AddComputeEngineInstanceGroupAddInstances()
+	s.query.AddCloudBuildActivityQuery()
 }
 
 var cdExecutionNamePattern = regexp.MustCompile(`^defang-cd-[a-z0-9]{5}$`)
 
-func getLogEntryParser(ctx context.Context, gcp *gcp.Gcp) func(entry *loggingpb.LogEntry) ([]*defangv1.TailResponse, error) {
+func getLogEntryParser(ctx context.Context, gcpClient *gcp.Gcp) func(entry *loggingpb.LogEntry) ([]*defangv1.TailResponse, error) {
 	envCache := make(map[string]map[string]string)
 	return func(entry *loggingpb.LogEntry) ([]*defangv1.TailResponse, error) {
 		if entry == nil {
 			return nil, nil
 		}
 
 		msg := entry.GetTextPayload()
+		if msg == "" && entry.GetJsonPayload() != nil {
+			msg = entry.GetJsonPayload().GetFields()["message"].GetStringValue()
```

Comment:
this feels like panic waiting to happen

Reply:
The whole structpb structure has sane nil function implementations, so
GetJsonPayload could return a nil *structpb.Struct
which when GetFields() is called on, returns nil map, 
which when indexed with a string key, returns a nil *structpb.Struct
which allows calling GetStringValue() and returns an empty string.